Fix SPIGOT-5824 Bukkit world-container is not used

BillyGalbreath opened the pull request:
Fixes https://hub.spigotmc.org/jira/browse/SPIGOT-5824

BillyGalbreath committed:
Fix SPIGOT-5824 Bukkit world-container is not used

TheMolkaPL commented on Spigot-Server-Patches/0543-Fix-SPIGOT-5824-Bukkit-world-container-is-not-used.patch:
```diff
@@ -0,0 +1,32 @@
++            } else {
++                try {
++                    file = new File(org.bukkit.configuration.file.YamlConfiguration
++                            .loadConfiguration((File) optionset.valueOf("bukkit-settings"))
```
Use `load()` from YamlConfiguration object.

```java
public static YamlConfiguration loadConfiguration(@NotNull Reader reader) {
    Validate.notNull(reader, "Stream cannot be null");

    YamlConfiguration config = new YamlConfiguration();

    try {
        config.load(reader);
    } catch (IOException ex) {
        Bukkit.getLogger().log(Level.SEVERE, "Cannot load configuration from stream", ex);
    } catch (InvalidConfigurationException ex) {
        Bukkit.getLogger().log(Level.SEVERE, "Cannot load configuration from stream", ex);
    }

    return config;
}
```

When IOException or InvalidConfigurationException occurs it will result in NPE since `Bukkit.getServer()` returns null until CraftServer sets `Bukkit.setServer(...)` in its constructor.

BillyGalbreath replied:
Good call ^_^

TheMolkaPL commented on Spigot-Server-Patches/0543-Fix-SPIGOT-5824-Bukkit-world-container-is-not-used.patch:
```diff
@@ -0,0 +1,32 @@
+-            File file = (File) optionset.valueOf("universe"); // CraftBukkit
++            // Paper start - fix SPIGOT-5824
++            File file;
++            if (optionset.has("universe")) {
```
Are you sure this isn't always `true`? `universe` is set to return a File by default.

BillyGalbreath replied:
Yes, `has` doesn't check the defaults, just what's `detected` to actually be set.

aikar committed:
Load config files early on

also fixes the ide commodore rewrite patch version

Prep for other PR's to be updated to use these as some options need to use some values earlier
